src/models/workspaces.ts

```diff
@@ -88,15 +92,15 @@ export const LearningLogWorkspaceModel = types
 
 export const PublishedWorkspaceModel = types
   .model("PublishedWorkspace", {
-    type: "published",
+    type: types.optional(types.literal(kPublicationID), kPublicationID),
     document: DocumentModel,
     createdAt: types.number,
     userId: types.string,
     groupId: types.string,
-    tool: types.maybe(types.undefined),
+    tool: types.optional(types.undefined, undefined),
```

Comment:
@kswenson We had talked about getting rid of this `undefined` field today, but I decided to keep it, changing the syntax to match the optional `literals` to clarify the intention. The `readOnly` field in the `workspace` component is useful, but we'd still get complaints whenever we try to access `workspace.tool` without this line. To me, this falls in the "use a default because it saves us some lines" category like the `mode` field.

Reply:
How about this alternative:
```javascript
export const PublishedWorkspaceModel = types
  .model("PublishedWorkspace", {
    type: types.optional(types.literal(kPublicationID), kPublicationID),
    document: DocumentModel,
    createdAt: types.number,
    userId: types.string,
    groupId: types.string,
    sectionId: types.string,
    groupUserConnections: types.map(types.boolean)
  })
  .views(self => ({
    get mode() {
      return "1-up";
    },
    get tool() {
      return undefined;
    }
  }));
```
We want the clients of the model to be able to refer to the `tool` and `mode` properties and to always get the same values, but we don't need to actually store those values with the model for that to be the case. The `type` field is different in that it has to be stored with the model for MST's union resolution to work, but that's not the case for the other fields.